server/src/main/scala/actors/RecoveryActorLike.scala

```diff
@@ -86,12 +86,21 @@ trait RecoveryActorLike extends PersistentActor with RecoveryLogging {
       playSnapshotMessage(ss)
 
     case RecoveryCompleted =>
-      log.info(s"Recovery complete. Took ${now().millisSinceEpoch - recoveryStartMillis}ms. $messagesPersistedSinceSnapshotCounter messages replayed.")
+      logRecoveryTime()
       postRecoveryComplete()
 
     case event: GeneratedMessage =>
       bytesSinceSnapshotCounter += event.serializedSize
       messagesPersistedSinceSnapshotCounter += 1
       playRecoveryMessage(event)
   }
+
+  private def logRecoveryTime(): Unit = {
+    val tookMs: MillisSinceEpoch = now().millisSinceEpoch - recoveryStartMillis
+    val message = s"Recovery complete. Took ${tookMs}ms. $messagesPersistedSinceSnapshotCounter messages replayed."
+    if (tookMs < 200L)
+      log.debug(message)
+    else
+      log.warn(s"SLOW $message")
```

Comment:
I like this idea - should we have an error threshold level as well?

Reply:
Good question. That's a though one. We already log an error on a timeout in fetching the data. I guess there's no harm in logging one for breaching a threshold. Maybe 10 seconds or something?

Reply:
I've added an error level log for 5s+